code/modules/mining/fulton.dm

```diff
@@ -0,0 +1,191 @@
+var/global/list/total_extraction_beacons = list()
+
+/obj/item/extraction_pack
+	name = "fulton extraction pack"
+	desc = "A balloon that can be used to extract equipment or personnel to a Fulton Recovery Beacon. Anything not bolted down can be moved. Link the pack to a beacon by using the pack in hand."
+	icon = 'icons/obj/fulton.dmi'
+	icon_state = "extraction_pack"
+	w_class = ITEMSIZE_NORMAL
+	var/obj/structure/extraction_point/beacon
+	var/list/beacon_networks = list("station")
+	var/uses_left = 3
+	var/can_use_indoors
+	var/safe_for_living_creatures = 1
+
+/obj/item/extraction_pack/examine()
+	. = ..()
+	usr.show_message("It has [uses_left] use\s remaining.", 1)
+
+/obj/item/extraction_pack/attack_self(mob/user)
+	var/list/possible_beacons = list()
+	for(var/B in global.total_extraction_beacons)
+		var/obj/structure/extraction_point/EP = B
+		if(EP.beacon_network in beacon_networks)
+			possible_beacons += EP
+
+	if(!possible_beacons.len)
+		to_chat(user, "There are no extraction beacons in existence!")
+		return
+
+	else
+		var/A
+
+		A = input("Select a beacon to connect to", "Balloon Extraction Pack", A) as null|anything in possible_beacons
+
+		if(!A)
+			return
+		beacon = A
+		to_chat(user, "You link the extraction pack to the beacon system.")
+
+/obj/item/extraction_pack/afterattack(atom/movable/A, mob/living/carbon/human/user, flag, params)
+	if(!beacon)
+		to_chat(user, "[src] is not linked to a beacon, and cannot be used.")
+		return
+	if(!can_use_indoors)
+		var/turf/T = get_turf(A)
+		if(T && !T.outdoors)
+			to_chat(user, "[src] can only be used on things that are outdoors!")
+			return
+	if(!flag)
+		return
+	if(!istype(A))
+		return
+	else
+		if(!safe_for_living_creatures && check_for_living_mobs(A))
+			to_chat(user, "[src] is not safe for use with living creatures, they wouldn't survive the trip back!")
+			return
+		if(!isturf(A.loc)) // no extracting stuff inside other stuff
+			return
+		if(A.anchored)
+			return
+		to_chat(user, "<span class='notice'>You start attaching the pack to [A]...</span>")
+		if(do_after(user,50,target=A))
+			to_chat(user, "<span class='notice'>You attach the pack to [A] and activate it.</span>")
+			/* No components, sorry. No convienence for you!
+			if(loc == user && istype(user.back, /obj/item/storage/backpack))
+				var/obj/item/storage/backpack/B = user.back
+				B.SendSignal(COMSIG_TRY_STORAGE_INSERT, src, user, FALSE, FALSE)
+			*/
+			uses_left--
+			if(uses_left <= 0)
```

Comment:
`if(--uses_left <= 0)`
Should the user just drop it when it runs out of charges, or should it delete itself?